services/src/api/model/operators.rs

```diff
@@ -91,10 +277,1148 @@ pub struct PlotResultDescriptor {
+#[derive(PartialEq, Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct GdalDatasetParameters {
+    pub file_path: PathBuf,
+    pub rasterband_channel: usize,
+    pub geo_transform: GdalDatasetGeoTransform, // TODO: discuss if we need this at all
+    pub width: usize,
+    pub height: usize,
+    pub file_not_found_handling: FileNotFoundHandling,
+    #[serde(default)]
+    //#[serde(with = "float_option_with_nan")]
+    pub no_data_value: Option<f64>,
+    pub properties_mapping: Option<Vec<GdalMetadataMapping>>,
+    // Dataset open option as strings, e.g. `vec!["UserPwd=geoengine:pwd".to_owned(), "HttpAuth=BASIC".to_owned()]`
+    pub gdal_open_options: Option<Vec<String>>,
+    // Configs as key, value pairs that will be set as thread local config options, e.g.
+    // `vec!["AWS_REGION".to_owned(), "eu-central-1".to_owned()]` and unset afterwards
+    // TODO: validate the config options: only allow specific keys and specific values
+    pub gdal_config_options: Option<Vec<(String, String)>>, // TODO: does not compile https://github.com/juhaku/utoipa/issues/330
+    #[serde(default)]
+    pub allow_alphaband_as_mask: bool,
+}
+
+impl From<geoengine_operators::source::GdalDatasetParameters> for GdalDatasetParameters {
+    fn from(value: geoengine_operators::source::GdalDatasetParameters) -> Self {
+        Self {
+            file_path: value.file_path,
+            rasterband_channel: value.rasterband_channel,
+            geo_transform: value.geo_transform.into(),
+            width: value.width,
+            height: value.height,
+            file_not_found_handling: value.file_not_found_handling.into(),
+            no_data_value: value.no_data_value,
+            properties_mapping: value
+                .properties_mapping
+                .map(|x| x.into_iter().map(Into::into).collect()),
+            gdal_open_options: value.gdal_open_options,
+            gdal_config_options: value.gdal_config_options,
+            allow_alphaband_as_mask: value.allow_alphaband_as_mask,
+        }
+    }
+}
+
+impl From<GdalDatasetParameters> for geoengine_operators::source::GdalDatasetParameters {
+    fn from(value: GdalDatasetParameters) -> Self {
+        Self {
+            file_path: value.file_path,
+            rasterband_channel: value.rasterband_channel,
+            geo_transform: value.geo_transform.into(),
+            width: value.width,
+            height: value.height,
+            file_not_found_handling: value.file_not_found_handling.into(),
+            no_data_value: value.no_data_value,
+            properties_mapping: value
+                .properties_mapping
+                .map(|x| x.into_iter().map(Into::into).collect()),
+            gdal_open_options: value.gdal_open_options,
+            gdal_config_options: value.gdal_config_options,
+            allow_alphaband_as_mask: value.allow_alphaband_as_mask,
+        }
+    }
+}
+
+// Utoipa does not support tuple types
```

Comment:
can you just extract the tuple types as a new type (that gets serialized the same way) and only implement the utoipa schema for them instead of the whole GdalDataasetParameters?

Reply:
[cc5d3a4](https://github.com/geo-engine/geoengine/pull/662/commits/cc5d3a44d356b9790d7b681c55d5f11082f13076)